### DISCUSSION TITLE
Changed to version 2.0.0

### DISCUSSION THREAD

**Discussion by MauricioGS99** [Joshuaalbert/jaxns#80] [Q&A] [answered]
Hello, I recently upgraded to version 2.0.0 and it seems lot of things changed. 

I'm building and sampling from Bayesian Networks generated at random. 

My perfectly working code (for the past version) is: 

def patient_1():
    disease = numpyro.sample("disease",dist.Bernoulli(0.4))
    treatment = numpyro.sample("treatment",dist.Bernoulli(0.7))
    list_variables=[disease,treatment]
    list_variables_no_treatment=[disease]
    q=np.random.randint(0,1)
    bloodpressure = numpyro.sample("bloodpressure",dist.Normal(non_linear_fn(jnp.array(random.sample(list_variables_no_treatment,np.random.randint(0,len(list_variables_no_treatment)))),q*treatment),1))
    list_variables.append(bloodpressure)
    if q==1:
        list_variables_no_treatment.append(bloodpressure)
     
    q=np.random.randint(0,1)
    weight = numpyro.sample("weight", dist.Normal(non_linear_fn(jnp.array(random.sample(list_variables_no_treatment,np.random.randint(0,len(list_variables_no_treatment)))),q*treatment), 1))
    list_variables.append(weight)
    if q==1:
        list_variables_no_treatment.append(weight)
    
    q=np.random.randint(0,1)
    heartattack=numpyro.sample("heartattack",dist.Normal(non_linear_fn(jnp.array(random.sample(list_variables_no_treatment,np.random.randint(1,len(list_variables_no_treatment)))),q*treatment),1))
    list_variables.append(heartattack)
    if q==1:
        list_variables_no_treatment.append(heartattack)
    
    variables=[]
    list_variables=[disease,treatment,bloodpressure,weight,heartattack]
    
    for i in range(5,math.floor(k/2)):
        q=np.random.randint(0,1)
        vi=numpyro.sample('variable'+str(i),dist.Normal(non_linear_fn(jnp.array(random.sample(list_variables,np.random.randint(0,len(list_variables)))),q*treatment), 1))
        variables.append(vi)
        list_variables.append(vi)
        if q==1:
            list_variables_no_treatment.append(vi)
    
    for i in range(math.floor(k/2)+1,k):
        q=np.random.randint(0,1)
        vi=numpyro.sample('variable'+str(i),dist.Normal(non_linear_fn(jnp.array(random.sample(list_variables_no_treatment,np.random.randint(0,len(list_variables_no_treatment)))),0*treatment), 1))
        variables.append(vi)
        list_variables.append(vi)
        list_variables_no_treatment.append(vi)
            
    q=np.random.randint(0,1)
    vi=numpyro.sample('variable'+str(k),dist.Normal(non_linear_fn(jnp.array(random.sample(list_variables,np.random.randint(0,k))),q*treatment), 1))
    variables.append(vi)
    list_variables.append(vi)
    if q==1:
        list_variables_no_treatment.append(vi)
    
    return list_variables  

and then I did
patient_1_kernel= NestedSampler(patient_1)
patient_1_kernel.run(jrn.PRNGKey(11493423))

The thing is, I don't know the log-likelihoods because the bayesian network is generated at random.

Any hints on how to proceed?

**Comment by Joshuaalbert** [comment, marked as answer]
Please try out 2.4.0. Documentation is better, and everything is improved.
